test/prefer-starts-ends-with.js

```diff
@@ -58,17 +47,93 @@ ruleTester.run('prefer-starts-ends-with', rule, {
 		'test()',
 		'test.test()',
 		'startWith("bar")',
-		'foo()()'
+		'foo()()',
+
+		...validRegex.map(re => `${re}.test(bar)`),
+		...validRegex.map(re => `bar.match(${re})`)
+	],
+	invalid: [
+		...invalidRegex.map(re => {
+			let messageId = MESSAGE_STARTS_WITH;
+			let method = 'startsWith';
+			let string = re.source;
+
+			if (string.startsWith('^')) {
+				string = string.slice(1);
+			} else {
+				messageId = MESSAGE_ENDS_WITH;
+				method = 'endsWith';
+				string = string.slice(0, -1);
+			}
+
+			return {
+				code: `${re}.test(bar)`,
+				output: `bar.${method}('${string}')`,
+				errors: [{messageId}]
+			};
+		}),
+		// Parenthesized
+		{
+			code: '/^b/.test(("a"))',
+			output: '("a").startsWith((\'b\'))',
+			errors: [{messageId: MESSAGE_STARTS_WITH}]
+		},
+		{
+			code: '(/^b/).test(("a"))',
+			output: '("a").startsWith((\'b\'))',
```

Comment:
string  `'b'` is  parenthesized because the  original `("a")` left the `()` when  replacing text, I'm not  going  to remove them.